corporate/views/remote_billing_page.py

```diff
@@ -99,7 +110,72 @@ def remote_realm_billing_finalize_login(
     except signing.BadSignature:
         raise JsonableError(_("Invalid billing access token."))
 
+    # Now we want to fetch (or create) the RemoteRealmBillingUser object implied
+    # by the IdentityDict. We'll use this:
+    # (1) If the user came here via just GET, we want to show them a confirmation
+    #     page with the relevant info details before finalizing login. If they wish
+    #     to proceed, they'll approve the form, causing a POST, bring us to case (2).
+    # (2) If the user came here via POST, we finalize login, using the info from the
+    #     IdentityDict to update the RemoteRealmBillingUser object if needed.
     remote_realm_uuid = identity_dict["remote_realm_uuid"]
+    remote_server_uuid = identity_dict["remote_server_uuid"]
+    try:
+        remote_server = get_remote_server_by_uuid(remote_server_uuid)
+        remote_realm = RemoteRealm.objects.get(uuid=remote_realm_uuid, server=remote_server)
+    except ObjectDoesNotExist:
+        # These should definitely still exist, since the access token was signed
+        # pretty recently. (And we generally don't delete these at all.)
+        raise AssertionError
+
+    user_dict = identity_dict["user"]
+
+    user_email = user_dict["user_email"]
+    user_full_name = user_dict["user_full_name"]
+    user_uuid = user_dict["user_uuid"]
+    remote_user, created = RemoteRealmBillingUser.objects.get_or_create(
+        defaults={"full_name": user_full_name, "email": user_email},
+        remote_realm=remote_realm,
+        user_uuid=user_uuid,
+    )
```

Comment:
I suppose we could create these only after a successful `POST`, with no entry being treated like `TOS_VERSION_BEFORE_FIRST_LOGIN`. 

I don't see a super strong reason to do it one way or the other, but I think the `RemoteZulipServer` version will probably need to do this after the form (since it gets the name/email from the form), and it feels like a cleaner consent model if we don't store the user data until after they explicitly agree to the ToS, so maybe we should move the creation down and just have a `remote_user = RemoteRealmBillingUser.objects.get` here?

Reply:
Oh yeah, it seems much better to only store user info after ToS consent. Restructured this bit so that `get_or_create` only happens in the POST codepath after consent is verified.